### PR TITLE
feat: add feature flag for guest room link (AR-2452)

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### description
In this PR, I am adding a feature flag to enable/disable guest room link from admin console.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

N/A

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt [hunk 1]
```diff
@@ -81,7 +81,33 @@ internal class FeatureConfigEventReceiverImpl internal constructor(
...
+                handleGuestRoomLinkFeatureConfig(event.model.status)
+            }
+
             is Event.FeatureConfig.UnknownFeatureUpdated -> kaliumLogger.w("Ignoring unknown feature config update")
         }
     }
+
+    private fun handleGuestRoomLinkFeatureConfig(status: Status) {
+        if (!kaliumConfigs.guestRoomLink) {
+            userConfigRepository.setGuestRoomStatus(false, null)
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt:94]: Could you explain a bit more to me here why we are setting it to false if we don't have `kaliumConfigs.guestRoomLink`?

**Review by alexandreferris — COMMENTED** [review]
Looking great!

I just have one comment about naming that for me it doesn't properly state what is actually returning and one question on how it works 😅

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt (hunk 1 above)
Comment (ohassine) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt:94]: `kaliumConfigs` will return the default value from `default.json`, if it's false then we disable the feature

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/SyncFeatureConfigsUseCase.kt [hunk 2]
```diff
@@ -50,7 +50,7 @@ internal class SyncFeatureConfigsUseCaseImpl(
     private val userConfigRepository: UserConfigRepository,
     private val featureConfigRepository: FeatureConfigRepository,
     private val isFileSharingEnabledUseCase: IsFileSharingEnabledUseCase,
-    private val isGuestRoomLinkFeatureEnabled: IsGuestRoomLinkFeatureEnabledUseCase,
+    private val isGuestRoomLinkFeatureEnabled: GetGuestRoomLinkFeatureStatusUseCase,
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/SyncFeatureConfigsUseCase.kt:53]:
```suggestion
    private val getGuestRoomLinkFeatureEnabled: GetGuestRoomLinkFeatureStatusUseCase,
```

**Review by alexandreferris — COMMENTED** [review]
just 2 more naming changing and all good 👌🏻

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**